### ISSUE TITLE
[Bug]: proxy: ollama backend returns empty response without streaming

### ISSUE THREAD

**Issue by emsi** [BerriAI/litellm#1119]
### What happened?

I'm running litellm proxy with:
```
litellm --model ollama/deepseek-coder:33b --debug --api_base http://localhost:11434 --drop_params --add_function_to_prompt
```

When I run:
```
curl --location 'http://127.0.0.1:8000/chat/completions' --header 'Content-Type: application/json' --data ' {"model": "gpt-3.5-turbo", "messages": [{"role": "user", "content": "what llm are you"} ], "stream": true }'
```
I get the proper answer, however when requesting without streaming I get:
```
curl --location 'http://127.0.0.1:8000/chat/completions' --header 'Content-Type: application/json' --data ' {"model": "gpt-3.5-turbo", "messages": [{"role": "user", "content": "what llm are you"} ], "stream": false }'

{"id":"chatcmpl-eefe69e3-ecd1-4ce2-b6ce-4786eeda8bc1","choices":[{"finish_reason":"stop","index":0,"message":{"content":"","role":"assistant"}}],"created":1702487580,"model":"ollama/deepseek-coder:33b","object":"chat.completion","system_fingerprint":null,"usage":{"prompt_tokens":5,"total_tokens":5}}
```
The response `content` is empty.
I have verified talking directly to ollama back-end that both streaming and non-streaming requests are handled properly.

The non-streaming response from ollama for reference:

```
curl -X POST http://localhost:11434/api/generate -d '{"model": "deepseek-coder:33b", "prompt": "what llm are you", "stream": false}'

{"model":"deepseek-coder:33b","created_at":"2023-12-13T17:15:05.074893341Z","response":"I'm sorry, but as an AI Programming Assistant, my function is to assist with computer science and programming-related queries. I don't have personal opinions or emotions, so I can't be classified in terms of LLM (Large Language Models).\n","done":true,"context":[2042,417,274,20926,14244,20391,11,26696,254,20676,30742,339,8589,2008,11,6908,457,20676,30742,7958,11,285,340,885,3495,4301,4512,276,4531,8214,13,1487,4636,2223,13143,4301,11,5411,285,13936,4447,11,285,746,2159,12,13517,250,8214,4301,11,340,540,20857,276,3495,13,185,185,13518,3649,3475,25,185,5003,1703,76,417,340,185,13518,21289,25,185,185,40,6,76,11572,11,545,372,274,20926,8285,4572,23457,11,597,1155,317,276,4737,365,4531,8214,285,14244,12,10664,12780,13,303,1115,6,83,463,3665,15725,409,19425,11,555,303,482,6,83,330,23991,279,3757,280,27480,44,334,27180,18226,4851,1289,628,185],"total_duration":1564918207,"load_duration":1850201,"prompt_eval_count":75,"prompt_eval_duration":265361000,"eval_count":56,"eval_duration":1292984000}
```

**Comment by ghost** [comment]
That's odd. Thanks for filing this @emsi, will take a look at this once the gemini-pro release is done.

**Comment by ghost** [comment]
Hey @emsi this is fixed here - https://github.com/BerriAI/litellm/commit/7b8851cce5ea36da90fbc39d37a16e8ced9a2621

should be live in v`1.14.0`

Can you confirm it works?

**Comment by emsi** [comment]
> Hey @emsi this is fixed here - [7b8851c](https://github.com/BerriAI/litellm/commit/7b8851cce5ea36da90fbc39d37a16e8ced9a2621)
> 
> should be live in v`1.14.0`
> 
> Can you confirm it works?

I checked using the current git head and it does not work.
The only difference is that it returns:
`"message":{"content":"\n","role":"assistant"}}]`
rather than empty content. 

Please take a look at proposed workaround and my other comments. The current implementation seems to have general problem with the way it handles async responses. If the client code (as a ollama client in litellm) finishes too early it fails to fetch the rest of the response because the client is not there long enough. Passing `'stream': false` in the request addresses that as the server does return the response in one shot. Despite this being only a workaround it should not have any adverse effects as the the aiohttp request is awaited and it does not block litellm thread.

**Comment by ghost** [comment]
@emsi do you have a test script i can run to repro this? 

I ran this test script with mistral-openorca and wasn't able to repro the issue (which i think is related to model responses taking longer than asyncio default timeouts)

```python
from openai import AsyncOpenAI
import random, uuid, aiohttp, traceback
import time, asyncio, json

#### OLLAMA API #### 
litellm_client = AsyncOpenAI(
    api_key="sk-1234",
    base_url="http://0.0.0.0:8000"
)

async def litellm_completion_streaming(i):
    print("in litellm_completion")

    stream = await litellm_client.chat.completions.create(
        model="mistral-openorca",
        messages=[{"role": "user", "content": f"Alexander the Great, born in 356 BC, was a brilliant military tactician and one of the most influential figures in ancient history. He ascended to the throne of Macedonia after the assassination of his father, King Philip II. Alexander's empire stretched from Greece to Egypt and as far east as India, making him one of the most successful conquerors of all time. He implemented various innovative military strategies, including the phalanx formation, and was known for his ability to inspire his troops with his leadership. Additionally, Alexander fostered cultural exchanges and the diffusion of Greek culture throughout his empire, resulting in a significant impact on the development of civilization in the regions he conquered. Question: How did Alexander's empire promote the spread of Greek culture, and what were some of the lasting effects of this assimilation process? {time.time()}"}],
        stream=True
        )
    async for chunk in stream:
        if chunk.choices[0].delta.content is not None:
            print(chunk.choices[0].delta.content)

async def litellm_completion(i):
    print("in litellm_completion")

    response = await litellm_client.chat.completions.create(
        model="mistral-openorca",
        messages=[{"role": "user", "content": f"what llm are you? {time.time()}"}],
        max_tokens=50
        )
    print(f"response: {response}")
    return response

async def loadtest_fn():
    start = time.time()
    n = 4 # Number of concurrent tasks
    # tasks = [create_chat_completion() for _ in range(n)]
    tasks = [litellm_completion(i) for i in range(n)]
    chat_completions = await asyncio.gather(*tasks)
    print(f"chat completions: {chat_completions}")
    successful_completions = [c for c in chat_completions if c is not None]
    print(n, time.time() - start, len(successful_completions))

# Run the event loop to execute the async function
asyncio.run(loadtest_fn())

```

**Comment by ghost** [comment]
Created a new issue to track this testing discussion - https://github.com/BerriAI/litellm/issues/1139

@emsi any help would be great!

**Comment by ghost** [comment]
I think i'm able to see this issue with the larger model. My normal ollama call is taking a 1min+ 

<img width="1077" alt="Screenshot 2023-12-14 at 10 21 52 AM" src="https://github.com/BerriAI/litellm/assets/17561003/6b7cbff5-eea4-400d-97e3-0493d377bf67">


@emsi

**Comment by emsi** [comment]
Notes:

1. ~~It's more apparent with fast models.~~ (that's probably not true but clearly the model I experienced it with was answering in under 40 seconds so it is NOT related to 60s timeout at the start)
2. Make sure the model is loaded with ollma first so a "prefetch" call to ollama may be necessary to make sure the model is loaded.
3. I'm experiencing it with `deepseek-coder:6.7b-instruct-q8_0` and 24GB VRAM GPU though I see ollama allocatede only `8600MiB` so perhaps a 12GB card should be enough.

**Comment by ghost** [comment]
tracking - https://github.com/BerriAI/litellm/issues/1139
